sos/collector/sosnode.py

```diff
@@ -0,0 +1,914 @@
+
+    def _sanitize_log_msg(self, msg):
+        """Attempts to obfuscate sensitive information in log messages such as
+        passwords"""
+        reg = r'(?P<var>(pass|key|secret|PASS|KEY|SECRET).*?=)(?P<value>.*?\s)'
+        return re.sub(reg, r'\g<var>****** ', msg)
+
+    def ui_msg(self, msg):
+        """Format a ui message that includes host name and formatting"""
+        self.ui_log.info(self._fmt_msg(msg))
+
+    def log_info(self, msg):
+        """Used to print and log info messages"""
+        caller = inspect.stack()[1][3]
+        lmsg = '[%s:%s] %s' % (self._hostname, caller, msg)
+        self.soslog.info(lmsg)
+
+    def log_error(self, msg):
+        """Used to print and log error messages"""
+        caller = inspect.stack()[1][3]
+        lmsg = '[%s:%s] %s' % (self._hostname, caller, msg)
+        self.soslog.error(lmsg)
+
+    def log_debug(self, msg):
+        """Used to print and log debug messages"""
+        msg = self._sanitize_log_msg(msg)
+        caller = inspect.stack()[1][3]
+        msg = '[%s:%s] %s' % (self._hostname, caller, msg)
+        self.soslog.debug(msg)
+
+    def get_hostname(self):
+        """Get the node's hostname"""
+        sout = self.run_command('hostname')
+        self.hostname = sout['stdout'].strip()
+        self.log_info('Hostname set to %s' % self.hostname)
+
+    def _format_cmd(self, cmd):
+        """If we need to provide a sudo or root password to a command, then
+        here we prefix the command with the correct bits
+        """
+        if self.opts.become_root:
+            return "su -c %s" % quote(cmd)
+        if self.need_sudo:
+            return "sudo -S %s" % cmd
+        return cmd
+
+    def _fmt_output(self, output=None, rc=0):
+        """Formats the returned output from a command into a dict"""
+        if rc == 0:
+            stdout = output
+            stderr = ''
+        else:
+            stdout = ''
+            stderr = output
+        res = {'status': rc,
+               'stdout': stdout,
+               'stderr': stderr}
+        return res
+
+    def _load_sos_info(self):
+        """Queries the node for information about the installed version of sos
+        """
+        if self.host.container_version_command is None:
+            pkg = self.host.package_manager.pkg_version(self.host.sos_pkg_name)
+            ver = '.'.join(pkg['version'])
+            self.sos_info['version'] = ver
+        else:
+            # use the containerized policy's command
+            pkgs = self.run_command(self.host.container_version_command,
+                                    use_container=True, need_root=True)
+            ver = pkgs['stdout'].strip().split('-')[1]
+            if ver:
+                self.sos_info['version'] = ver
+        if 'version' in self.sos_info:
+            self.log_info('sos version is %s' % self.sos_info['version'])
+        else:
+            self.log_error('sos is not installed on this node')
```

Comment:
I dont know why, but if a remote system does not have `sos` package installed, then `self.connected = None` and report is not collected, BUT this error is _not_ printed - that is very confusing (and I dont get why)..